FreeRTOS_TCP_IP.c

```diff
@@ -3896,7 +3914,7 @@
         if( vSocketBind( pxNewSocket, &xAddress, sizeof( xAddress ), pdTRUE ) != 0 )
         {
             FreeRTOS_debug_printf( ( "TCP: Listen: new socket bind error\n" ) );
-            ( void ) FreeRTOS_closesocket( pxNewSocket );
+            vSocketClose( pxNewSocket );
```

Comment:
Why isn't this using `vSocketCloseNextTime()`?

Reply:
That is not necessary, because when this function returns pdFALSE the caller immediately
forgets the pxNewSocket.
```C
                    else if( prvTCPSocketCopy( pxNewSocket, pxSocket ) != pdFALSE )
                    {
                        /* The socket will be connected immediately, no time for the
                         * owner to setsockopt's, therefore copy properties of the server
                         * socket to the new socket.  Only the binding might fail (due to
                         * lack of resources). */
                        pxReturn = pxNewSocket;
                    }
                    else
                    {
                        /* Copying failed somehow. */
                    }

```

Reply:
@bernd-edlinger  wrote:
> 
> That is not necessary, because when this function returns pdFALSE the caller immediately
> forgets the pxNewSocket.

Good question, I also wondered, but the IP-task just failed to bind the new client socket `pxNewSocket` to a port number. The socket can be closed without delay because nobody has a reference to it, or will refer to it any more.